flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/relational/connection/JdbcConnectionPools.java

```diff
@@ -31,28 +33,40 @@ public class JdbcConnectionPools implements ConnectionPools<HikariDataSource, Jd
 
     private static JdbcConnectionPools instance;
     private final Map<ConnectionPoolId, HikariDataSource> pools = new HashMap<>();
-    private static JdbcConnectionPoolFactory jdbcConnectionPoolFactory;
+    private static Map<String, JdbcConnectionPoolFactory> poolFactorys = new HashMap<>();
 
     private JdbcConnectionPools() {}
 
     public static synchronized JdbcConnectionPools getInstance(
             JdbcConnectionPoolFactory jdbcConnectionPoolFactory) {
         if (instance == null) {
-            JdbcConnectionPools.jdbcConnectionPoolFactory = jdbcConnectionPoolFactory;
             instance = new JdbcConnectionPools();
         }
+        poolFactorys.put(jdbcConnectionPoolFactory.factoryIdentifier(), jdbcConnectionPoolFactory);
         return instance;
     }
 
     @Override
     public HikariDataSource getOrCreateConnectionPool(
-            ConnectionPoolId poolId, JdbcSourceConfig sourceConfig) {
+            ConnectionPoolId poolId,
+            JdbcSourceConfig sourceConfig,
+            String dataSourcePoolFactoryIdentifier) {
```

Comment:
I think we can put the identifier to the ConnectionPoolId, it's a little redundant to pass it here.